compiler/rustc_trait_selection/src/solve/assembly/structural_traits.rs

```diff
@@ -177,6 +177,11 @@ pub(in crate::solve) fn instantiate_constituent_tys_for_copy_clone_trait<'tcx>(
         | ty::Ref(_, _, Mutability::Not)
         | ty::Array(..) => Err(NoSolution),
 
+        // Check for anonymous adts.
+        ty::Adt(adt, generics) if adt.flags().contains(ty::AdtFlags::IS_ANONYMOUS) => {
+            Ok(adt.all_fields().map(|f| f.ty(ecx.tcx(), generics)).collect())
+        }
```

Comment:
Can you explain why the standard path for ADTs does not work?

Reply:
It was because that `#[derive(Clone)]` or other builtin proc macros didn't apply to the nested anonymous ADTs.

In a recent implementation or parsing (see #115732), those builtin deriving macros will process the nested anonymous ADTs so that here we don't need a special case for anonymous ADTs.

Reply:
So this can be removed now that the derive is in this PR?

Reply:
Sure